File: backend/src/main/java/com/expenseguard/receipt/entity/Receipt.java
package com.expenseguard.receipt.entity;

import com.expenseguard.auth.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;

/**
 * Domain Entity representing an uploaded receipt document and its OCR extraction metrics.
 */
@Entity
@Table(
        name = "receipts",
        indexes = {
                @Index(name = "idx_receipts_user_id", columnList = "user_id")
        }
)
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Receipt {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "file_name", nullable = false)
    private String fileName;

    @Column(name = "content_type", nullable = false)
    private String contentType;

    @Column(name = "storage_key", nullable = false, unique = true)
    private String storageKey;

    @Column(name = "file_size", nullable = false)
    private Long fileSize;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private ReceiptStatus status;

    @Column(name = "merchant_name")
    private String merchantName;

    @Column(name = "extracted_date")
    private LocalDate extractedDate;

    @Column(name = "extracted_amount", precision = 19, scale = 4)
    private BigDecimal extractedAmount;

    @Column(name = "confidence", precision = 5, scale = 4)
    private BigDecimal confidence;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        Instant now = Instant.now();
        this.createdAt = now;
        this.updatedAt = now;
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = Instant.now();
    }
}

File: mobile/src/types/receipt.ts
export type ReceiptStatus = 'UPLOADED' | 'PROCESSING' | 'PROCESSED' | 'FAILED' | 'CONFIRMED';

export interface ReceiptResponse {
  id: string;
  fileName: string;
  contentType: string;
  fileSize: number;
  status: ReceiptStatus;
  merchantName?: string | null;
  extractedDate?: string | null;
  extractedAmount?: number | null;
  confidence?: number | null;
  createdAt?: string;
  updatedAt?: string;
}

export interface ReceiptConfirmationRequest {
  merchantName: string;
  transactionDate: string; // YYYY-MM-DD
  amount: number;
}

export interface OcrResult {
  merchantName: string;
  transactionDate: string;
  amount: number;
  confidence: number;
  rawText?: string;
}

export interface SelectedImage {
  uri: string;
  fileName?: string;
  type?: string;
  fileSize?: number;
}
